plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+
+					continue;
+				}
+
+				$srcFilePath = "{$basePath}/{$data->bookCoverImage}";
+				if (!is_readable($srcFilePath)) {
+					$reason = __('plugins.importexport.csv.invalidCoverImageFilename', ['title' => $data->title]);
+					$failedRows = $this->processFailedRow($invalidRowsFile, $fieldsList, $reason, $failedRows);
+
+					continue;
+				}
+			}
+
+			$dbCategoryIds = $this->getCategoryDataForValidRow($data->categories, $pressId, $data->locale);
+			if (!$dbCategoryIds) {
+				$reason = __('plugins.importexport.csv.allCategoriesMustExists');
+				$failedRows = $this->processFailedRow($invalidRowsFile, $fieldsList, $reason, $failedRows);
+
+				continue;
+			}
+
+			// All requirements passed. Start processing from here.
+			$this->initializeStaticVariables();
+			$submission = $this->processSubmission($data, $pressId);
+			$publication = $this->processPublication($submission, $data, $press);
+			$this->processAuthors($data, $press->getContactEmail(), $submission->getId(), $publication, $userGroupId);
+
+			// Submission is done.  Create a publication format for it.
+			$extension = $this->_fileManager->parseFileExtension($data->filename);
+			$publicationFormatId = $this->processPublicationFormat($submission->getId(), $publication->getId(), $extension, $data);
+
+			$this->processPublicationDate($data->year, $publicationFormatId);
+
+			// Submission File.
+			$this->processPublicationFile(
+				$data,
+				$submission->getId(),
+				$filePath,
+				$extension,
+				$publicationFormatId,
+				$pressId,
+				$genreId
+			);
+
+			$this->processKeywords($data, $publication->getId());
+			$this->processSubjects($data, $publication->getId());
+
+			if ($data->bookCoverImage) {
+				$this->processBookCoverImage($data, $srcFilePath, $publication);
+			}
+
+			/** @var CategoryDAO $categoryDao */
+			$categoryDao = $this->getCachedDao('CategoryDAO');
+			foreach ($dbCategoryIds as $categoryId) {
+				$categoryDao->insertPublicationAssignment($categoryId, $publication->getId());
+			}
+
+			echo __('plugins.importexport.csv.import.submission', ['title' => $data->title]) . "\n";
+		}
+
+		if ($failedRows === 0) {
+			echo __('plugin.importexport.csv.allDataSuccessfullyImported', ['processedRows' => $processedRows]) . "\n\n";
+			unlink($csvForInvalidRowsName);
+		} else {
+			echo __('plugin.importexport.csv.seeInvalidRowsFile', ['processedRows' => $processedRows - $failedRows, 'failedRows' => $failedRows]) . "\n\n";
+		}
+	}
+
+	/**
+	 * Display the command-line usage information
+	 */
+	function usage($scriptName) {
+		echo __('plugins.importexport.csv.cliUsage', [
+			'scriptName' => $scriptName,
+			'pluginName' => $this->getName()
+		]) . "\n";
+	}
+
+	/**
+	 * Parse and validate initial command args
+	 *
+	 * @param string $scriptName
+	 * @param array $args
+	 * @return string[]
+	 */
+	private function parseCommandLineArguments($scriptName, $args) {
```

Comment:
I don't think it's important, but if you want to follow the pattern on the `stable-3_3_0`, the private methods should be prefixed with `_` too. On the `stable-3_4_0` and `main`, the pattern was changed and the `_` is not used anymore).

```suggestion
	private function parseCommandLineArguments($scriptName, $args) {
```